mne/time_frequency/psd.py

```diff
@@ -112,12 +111,209 @@ def _check_nfft(n, n_fft, n_overlap):
+           array of shape ([n_epochs], n_channels, n_times)
+        The data. If not an instance of Epochs or Raw,
+        sfreq must also be supplied.
+    fmin : float
+        Min frequency of interest
+    fmax : float
+        Max frequency of interest
+    tmin : float | None
+        Min time of interest
+    tmax : float | None
+        Max time of interest
+    n_fft : int
+        The length of the tapers ie. the windows. The smaller
+        it is the smoother are the PSDs. The default value is 256.
+        If ``n_fft > len(epochs.times)``, it will be adjusted down to
+        ``len(epochs.times)``.
+    n_overlap : int
+        The number of points of overlap between blocks. Will be adjusted
+        to be <= n_fft.
+        bandwidth : float
+        The bandwidth of the multi taper windowing function in Hz.
+    picks : array-like of int | None
+        The selection of channels to include in the computation.
+        If None, take all channels.
+    proj : bool
+        Apply SSP projection vectors. If inst is ndarray this is not used.
+    sfreq : float
+        The sampling frequency of the data. Required if inst is an array,
+        otherwise this is not used and sfreq is pulled from inst.
+    n_jobs : int
+        Number of CPUs to use in the computation.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    psds : ndarray, shape ([n_epochs], n_channels, n_freqs)
+        The power spectral densities. If Raw is provided,
+        then psds will be 2-D.
+    freqs : ndarray (n_freqs)
+        The frequencies.
+    """
+    from scipy.signal import welch
+
+    # Prep data
+    data, sfreq = _check_psd_data(inst, tmin, tmax, picks, proj, sfreq)
+    dshape = data.shape[:-1]
+    n_times = data.shape[-1]
+
+    # This will return the same object if len(dshape) == 1
+    data = data.reshape(np.product(dshape), n_times)
+
+    # Prep the PSD
+    n_fft, n_overlap = _check_nfft(n_times, n_fft, n_overlap)
+    win_size = n_fft / float(sfreq)
+    logger.info("Effective window size : %0.3f (s)" % win_size)
+    freqs = np.arange(n_fft // 2 + 1, dtype=float) * (sfreq / n_fft)
+    freq_mask = (freqs >= fmin) & (freqs <= fmax)
+    freqs = freqs[freq_mask]
+    n_freqs = len(freqs)
+
+    # Parallelize across first N-1 dimensions
+    psds = np.empty(data.shape[:-1] + (freqs.size,))
+    parallel, my_pwelch, n_jobs = parallel_func(_pwelch, n_jobs=n_jobs,
+                                                verbose=verbose)
+    data_splits = np.array_split(data, n_jobs)
+    f_psd = parallel(my_pwelch(d, noverlap=n_overlap, nfft=n_fft,
+                     fs=sfreq, freq_mask=freq_mask,
+                     welch_fun=welch)
+                     for d in data_splits)
+
+    # Combining/reshaping to original data shape
+    psds = np.concatenate(f_psd, axis=0)
+    psds = psds.reshape(np.hstack([dshape, n_freqs]))
```

Comment:
good idea on reshaping the data, but it wouldn't work here if we support both 2d or 3d arrays. If we don't want this behavior in here, then I could bump it up into `psd_welch` instead of being in the private function